packages/qdrant-loader-core/src/qdrant_loader_core/llm/providers/gemini.py

```
@@ -0,0 +1,350 @@
+    LLMError,
+    RateLimitedError,
+    ServerError,
+)
+from ..errors import TimeoutError as LLMTimeoutError
+from ..settings import LLMSettings
+from ..types import ChatClient, EmbeddingsClient, LLMProvider, TokenCounter
+
+logger = LoggingConfig.get_logger(__name__)
+
+
+def _map_gemini_exception(exc: Exception) -> LLMError:
```

Ruff (0.15.14)

[warning] Too many return statements (7 > 6)

(PLR0911)